src/ImmersedBoundaries/partial_cell_immersed_boundaries.jl

```diff
@@ -0,0 +1,91 @@
+using Adapt
+using CUDA: CuArray
+using Oceananigans.Fields: fill_halo_regions!
+using Oceananigans.Architectures: arch_array
+
+import Oceananigans.TurbulenceClosures: ivd_upper_diagonal,
+                                        ivd_lower_diagonal
+
+#####
+##### PartialCellBottom
+#####
+
+struct PartialCellBottom{B, E} <: AbstractGridFittedBoundary
+    bottom_height :: B
+    minimum_fractional_partial_Δz :: E
+end
+
+"""
+    PartialCellBottom(bottom, minimum_height)
+
+Return an immersed boundary...
+"""
+PartialCellBottom(bottom_height; minimum_fractional_partial_Δz=0.1) =
+    PartialCellBottom(bottom_height, minimum_fractional_partial_Δz)
+
+@inline get_bottom_height(i, j, k, grid, bottom_height::AbstractArray) = @inbounds bottom_height[i, j]
+
+@inline function get_bottom_height(i, j, k, underlying_grid, bottom_height)
+    x, y, z = node(c, c, f, i, j, k, underlying_grid)
+    return bottom_height(x, y)
+end
+
+"""
+
+        --x--
+          ∘   k+1
+    k+1 --x--    ↑      <- node z
+          ∘   k  | Δz
+      k --x--    ↓
+      
+Criterion is h >= z - ϵ Δz
+
+"""
+@inline function is_immersed(i, j, k, underlying_grid, ib::PartialCellBottom)
+    # Face node above current cell
+    x, y, z = node(c, c, f, i, j, k+1, underlying_grid)
+    h = get_bottom_height(i, j, k, underlying_grid, ib.bottom_height)
+    return h >= z
+end
+
+const PCIBG = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:PartialCellBottom}
+
+bottom_cell(i, j, k, ibg::PCIBG) = !is_immersed(i, j, k, ibg.grid, ibg.immersed_boundary) & is_immersed(i, j, k-1, ibg.grid, ibg.immersed_boundary)
+
+@inline function Δzᶜᶜᶜ(i, j, k, ibg::PCIBG)
+    underlying_grid = ibg.grid
+    ib = ibg.immersed_boundary
+    # Get node at face above
+    x, y, z = node(c, c, c, i, j, k+1, underlying_grid)
```

Comment:
@fadaie91 this is the line you believe has a typo right? Can you paste your sketch into this PR?

Reply:
@fadaie91 : Can you confirm we are proposing that in line 59, we should have '(c, c, f)'?  

That would certainly work with the comment but it's been a while so I might be misremembering.

Reply:
Yes we should have (c, c, f)